pytensor/link/jax/dispatch/scan.py

```diff
@@ -151,11 +158,18 @@ def get_partial_traces(traces):
                 + op.outer_nitsot(outer_inputs)
             )
             partial_traces = []
-            for init_state, trace, buffer in zip(init_states, traces, buffers):
+            for init_state, trace, add_batchdim, buffer in zip(
+                init_states, traces, add_batchdim_flags, buffers
+            ):
                 if init_state is not None:
                     # MIT-SOT and SIT-SOT: The final output should be as long as the input buffer
+                    if add_batchdim:
```

Comment:
Is this enough (without the need for the flag variables)?

```python
trace = jnp.atleast_1d(trace)
init_state = jnp.expand_dims(init_state, range(trace.ndim - init_state.ndim))
full_trace = jnp.concatenate([init_state, trace], axis=0,)
```

Reply:
Nice, I'll give it a try.

Reply:
Works perfectly